### PR TITLE
Update event subscriber

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- support sending cloud events
- support basic counter usage
- use the updated image in upgrade test
- allow configuration of the event-subscriber image in the `values.yaml` (for upgrade test)

**Related issue(s)**

#8483
#8578

### CODE REVIEW

File: tools/event-subscriber/README.md [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
 This is a sample subscriber app that simulates an event subscriber via exposing a set of APIs which can be used to receive and list events. It is meant to be only used in tests.
 
 
+## Endpoints
+
+| Endpoint                                     | Method | Usage                                                                                             |
+|-------------------------------------------- |------ |------------------------------------------------------------------------------------------------- |
+| `/ce`                                      | POST   | stores the received cloudevent                                                                    |
+| `/ce`                                      | GET    | return list of all received cloudevents                                                           |
+| `/ce/<uuid>`                               | GET    | return cloud event with matching uuid. Returns 200 on success, 204 if no event was found          |
```
[the remark this answers is not in the document]
Comment (k15r) [reply tools/event-subscriber/README.md:12]: and how would you differentiate then from endpoint not found?

**Comment by nachtmaar** [comment]
> Nice idea and good job!
> 
> I wonder if we should have some tests. It may sound weird because the tool is used inside tests, but debugging e2e-external-solution takes a lot of time, so I think having tests makes sense to save e2e debug time in case then there is a bug in the tool itself. What is your take on tests ?

We discussed it and decided on creating a follow-up issue: https://github.com/kyma-project/kyma/issues/8578

File: tools/event-subscriber/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nachtmaar) [reply tools/event-subscriber/README.md:12]: As discussed: 404 or 204. Both works.

File: tools/event-subscriber/cmd/subscriber.go [hunk 2]
```diff
@@ -2,24 +2,153 @@ package main
...
+func getAllCE(w http.ResponseWriter, r *http.Request) {
+	mu.Lock()
+	defer mu.Unlock()
+	events := make([]cloudevents.Event, 0)
+	for _, event := range receivedCEs {
+		events = append(events, event)
+	}
+	log.Infof("Getting all CE: %v", events)
+
+	if err := json.NewEncoder(w).Encode(events); err != nil {
```
[the remark this answers is not in the document]
Comment (k15r) [reply tools/event-subscriber/cmd/subscriber.go:127]: map vs slice

File: tools/event-subscriber/README.md [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+| `/ce/<source>/<type>/<event-type-version>` | GET    | return all events matching source/type/version. Returns 200 on success, 204 if no event was found |
```
Comment (bszwarc) [inline tools/event-subscriber/README.md:13]:
```suggestion
| `/ce/<source>/<type>/<event-type-version>` | GET    | Returns all CloudEvents matching the source/type/version. Returns `200` on success, `204` if no event was found. |
```

File: tools/event-subscriber/README.md [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
+| `/`                                        | POST   | increase counter of received requests. Cloudevents sent to /ce do not increase this counter       |
```
Comment (bszwarc) [inline tools/event-subscriber/README.md:14]:
```suggestion
| `/`                                        | POST   | Increases the counter of received requests. CloudEvents sent to `/ce` do not increase this counter.     |
```

File: tools/event-subscriber/README.md [hunk 5]
```diff
@@ -1,4 +1,22 @@
...
+| `/`                                        | GET    | return current counter value                                                                      |
```
Comment (bszwarc) [inline tools/event-subscriber/README.md:15]:
```suggestion
| `/`                                        | GET    | Returns the current counter value.                                                                     |
```

File: tools/event-subscriber/README.md [hunk 6]
```diff
@@ -1,4 +1,22 @@
...
+| `/`                                        | DELETE | reset counter and list of received cloudevents                                                    |
```
Comment (bszwarc) [inline tools/event-subscriber/README.md:16]:
```suggestion
| `/`                                        | DELETE | Resets the counter and clears the list of received CloudEvents.                                                    |
```

File: tools/event-subscriber/README.md [hunk 7]
```diff
@@ -1,4 +1,22 @@
...
+
+## Commandline
+
+| Flag                   | Usage          |
+|---------------------- |-------------- |
+| `--port <int>` | listen on port |
```
Comment (bszwarc) [inline tools/event-subscriber/README.md:22]:
```suggestion
| `--port <int>` | Listens on o a provided port.|
```

File: tools/event-subscriber/cmd/subscriber.go [hunk 8]
```diff
@@ -2,24 +2,166 @@ package main
...
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	events := make([]cloudevents.Event, 0)
+	for _, event := range s.receivedCEs {
+		events = append(events, event)
+	}
+	log.Infof("Getting all CE: %v", events)
+
+	if err := json.NewEncoder(w).Encode(events); err != nil {
+		log.Errorf("Error during getAllCE: %v", err)
```
Comment (sayanh) [inline tools/event-subscriber/cmd/subscriber.go:128]:
```suggestion
		log.Errorf("error during getAllCE: %v", err)
```

File: tools/event-subscriber/cmd/subscriber.go [hunk 9]
```diff
@@ -2,24 +2,166 @@ package main
...
+	uuid := vars["uuid"]
+	ce, exists := s.receivedCEs[uuid]
+	if !exists {
+		log.Infof("Checking for uuid: %v. Not found", uuid)
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+	log.Infof("Checking for uuid: %v. found: %v", uuid, ce)
+	if err := json.NewEncoder(w).Encode(ce); err != nil {
+		log.Errorf("Error during checkCEbyUUID: %v", err)
```
Comment (sayanh) [inline tools/event-subscriber/cmd/subscriber.go:146]:
```suggestion
		log.Errorf("error during checkCEbyUUID: %v", err)
```

File: tools/event-subscriber/cmd/subscriber.go [hunk 10]
```diff
@@ -2,24 +2,166 @@ package main
...
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (s *SubscriberState) checkCounter(w http.ResponseWriter, r *http.Request) {
+	response := counterResponse{Counter: int(s.counter)}
+	log.Infof("Checking counter: %v", response)
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		log.Errorf("Error during checkCounter: %v", err)
```
Comment (sayanh) [inline tools/event-subscriber/cmd/subscriber.go:156]:
```suggestion
		log.Errorf("error during checkCounter: %v", err)
```

File: tools/event-subscriber/cmd/subscriber_test.go [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+	req = httptest.NewRequest("GET", "/", nil)
+	w = httptest.NewRecorder()
+	router.ServeHTTP(w, req)
+	err = json.Unmarshal(w.Body.Bytes(), &cr)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, cr.Counter)
+}
+
+func Test_SendCE(t *testing.T) {
+	type tabletest struct {
```
Comment (sayanh) [inline tools/event-subscriber/cmd/subscriber_test.go:91]:
```suggestion
	type tableTest struct {
```

**Pull request merged**